src/vs/workbench/electron-browser/workbench.ts

```diff
@@ -1313,6 +1323,22 @@ export class Workbench implements IPartService {
 		}
 	}
 
+	public isCenteredLayoutActive(): boolean {
+		return this.centeredLayoutActive;
+	}
+
+	public toggleCenteredLayout(): void {
+		this.centeredLayoutActive = !this.centeredLayoutActive;
+
+		if (this.centeredLayoutActive) {
```

Comment:
I would replace this if statement with
`this.storageService.store(Workbench.centeredLayoutActiveStorageKey, this.centeredLayoutActive, StorageScope.GLOBAL);`

Seems more straighforwared to me.
It seems like we were doing this clunky business in zen mode so you took it from there.

Reply:
done.

Reply:
Actually you doing this 'clunky' business all the time for stored boolean values.